### PR TITLE
openai[patch]: update openai params

### PR DESCRIPTION
**Pull request by Samoed**
**Description:** Explicitly add parameters from openai API

- [X] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

### CODE REVIEW

**Review by efriis — COMMENTED** [review]
just one comment on validation, otherwise lgtm!

**Comment by Samoed** [comment]
@efriis Fixed

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 1]
```diff
@@ -317,10 +317,26 @@ class BaseChatOpenAI(BaseChatModel):
         None."""
     max_retries: int = 2
     """Maximum number of retries to make when generating."""
+    presence_penalty: float = 0
+    """Penalizes repeated tokens."""
+    frequency_penalty: float = 0
+    """Penalizes repeated tokens according to frequency."""
```
Comment (efriis) [inline libs/partners/openai/langchain_openai/chat_models/base.py:323]: sorry another one. Would it make sense to make these optional, and not pass them when they're None? Then we rely on any defaults openai sets.

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 2]
```diff
@@ -317,10 +317,26 @@ class BaseChatOpenAI(BaseChatModel):
...
+    """Number of most likely tokens to return at each token position, each with
+     an associated log probability. `logprobs` must be set to true 
+     if this parameter is used."""
+    logit_bias: Optional[Dict[int, int]] = None
+    """Modify the likelihood of specified tokens appearing in the completion."""
     streaming: bool = False
     """Whether to stream the results or not."""
     n: int = 1
     """Number of chat completions to generate for each prompt."""
+    top_p: float = 1.0
```
Comment (efriis) [inline libs/partners/openai/langchain_openai/chat_models/base.py:338]: same here

**Review by efriis — COMMENTED** [review]
can we add some unit and integration tests that test these?

@baskaryan thoughts on adding these to help people with code completion, vs just documenting them in the class docstring because they would get passed in as `model_kwargs`?

**Review by efriis — APPROVED** [review]

File: libs/partners/openai/langchain_openai/chat_models/base.py [hunk 3]
```diff
@@ -319,10 +319,26 @@ class BaseChatOpenAI(BaseChatModel):
...
+    """Number of most likely tokens to return at each token position, each with
+     an associated log probability. `logprobs` must be set to true 
+     if this parameter is used."""
+    logit_bias: Optional[Dict[int, int]] = None
+    """Modify the likelihood of specified tokens appearing in the completion."""
     streaming: bool = False
     """Whether to stream the results or not."""
     n: int = 1
     """Number of chat completions to generate for each prompt."""
+    top_p: Optional[float] = None
```
Comment (efriis) [inline libs/partners/openai/langchain_openai/chat_models/base.py:340]: confirmed None results in 1 in response

**Review by efriis — APPROVED** [review]

**Pull request merged**